File: data/src/main/java/com/github/gmetal/data/entity/TvShowEntity.java
package com.github.gmetal.data.entity;

import com.bluelinelabs.logansquare.annotation.JsonField;
import com.bluelinelabs.logansquare.annotation.JsonObject;

import java.util.ArrayList;
import java.util.List;

@JsonObject
public class TvShowEntity {

    @JsonField(name = "backdrop_path")
    public String backdropPath;
    @JsonField(name = "created_by")
    public List<CreatedByEntity> createdByEntity = new ArrayList<>();
    @JsonField(name = "episode_run_time")
    public List<Integer> episodeRunTime = new ArrayList<>();
    @JsonField(name = "first_air_date")
    public String firstAirDate;
    @JsonField(name = "genreEntities")
    public List<GenreEntity> genreEntities = new ArrayList<>();
    @JsonField(name = "homepage")
    public String homepage;
    @JsonField(name = "id")
    public Integer id;
    @JsonField(name = "in_production")
    public Boolean inProduction;
    @JsonField(name = "languages")
    public List<String> languages = new ArrayList<>();
    @JsonField(name = "last_air_date")
    public String lastAirDate;
    @JsonField(name = "name")
    public String name;
    @JsonField(name = "networkEntities")
    public List<NetworkEntity> networkEntities = new ArrayList<>();
    @JsonField(name = "number_of_episodes")
    public Integer numberOfEpisodes;
    @JsonField(name = "number_of_seasons")
    public Integer numberOfSeasons;
    @JsonField(name = "origin_country")
    public List<String> originCountry = new ArrayList<>();
    @JsonField(name = "original_language")
    public String originalLanguage;
    @JsonField(name = "original_name")
    public String originalName;
    @JsonField(name = "overview")
    public String overview;
    @JsonField(name = "popularity")
    public Double popularity;
    @JsonField(name = "poster_path")
    public String posterPath;
    @JsonField(name = "production_companies")
    public List<ProductionCompanyEntity> productionCompanies = new ArrayList<>();
    @JsonField(name = "seasonEntities")
    public List<SeasonEntity> seasonEntities = new ArrayList<>();
    @JsonField(name = "status")
    public String status;
    @JsonField(name = "type")
    public String type;
    @JsonField(name = "vote_average")
    public Float voteAverage;
    @JsonField(name = "vote_count")
    public Integer voteCount;
}

File: data/src/main/java/com/github/gmetal/data/entity/TvShowEntity.kt
package com.github.gmetal.data.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class TvShowEntity {

    @SerialName("backdrop_path")
    var backdropPath: String? = null
    @SerialName("created_by")
    var createdByEntity: List<CreatedByEntity> = ArrayList()
    @SerialName("episode_run_time")
    var episodeRunTime: List<Int> = ArrayList()
    @SerialName("first_air_date")
    var firstAirDate: String? = null
    @SerialName("genreEntities")
    var genreEntities: List<GenreEntity> = ArrayList()
    @SerialName("homepage")
    var homepage: String? = null
    @SerialName("id")
    var id: Int? = null
    @SerialName("in_production")
    var inProduction: Boolean? = null
    @SerialName("languages")
    var languages: List<String> = ArrayList()
    @SerialName("last_air_date")
    var lastAirDate: String? = null
    @SerialName("name")
    var name: String? = null
    @SerialName("networkEntities")
    var networkEntities: List<NetworkEntity> = ArrayList()
    @SerialName("number_of_episodes")
    var numberOfEpisodes: Int? = null
    @SerialName("number_of_seasons")
    var numberOfSeasons: Int? = null
    @SerialName("origin_country")
    var originCountry: List<String> = ArrayList()
    @SerialName("original_language")
    var originalLanguage: String? = null
    @SerialName("original_name")
    var originalName: String? = null
    @SerialName("overview")
    var overview: String? = null
    @SerialName("popularity")
    var popularity: Double? = null
    @SerialName("poster_path")
    var posterPath: String? = null
    @SerialName("production_companies")
    var productionCompanies: List<ProductionCompanyEntity> = ArrayList()
    @SerialName("seasons")
    var seasonEntities: List<SeasonEntity> = ArrayList()
    @SerialName("status")
    var status: String? = null
    @SerialName("type")
    var type: String? = null
    @SerialName("vote_average")
    var voteAverage: Float? = null
    @SerialName("vote_count")
    var voteCount: Int? = null
}
